bot/code_coverage_bot/codecov.py

```diff
@@ -146,27 +142,33 @@ def build_suites(self):
                 artifacts, source_dir=self.repo_dir, out_format="covdir"
             )
 
-            # Then upload on GCP
-            report = json.loads(output)
+            # Write output on FS
+            path = os.path.join(self.reports_dir, f"{platform}.{suite}.json")
+            with open(path, "wb") as f:
+                f.write(output)
+
+            reports[(platform, suite)] = path
+
+        return reports
+
+    def upload_reports(self, reports):
+        """
+        Upload all provided covdir reports on GCP
+        """
+        for (platform, suite), path in reports.items():
+            report = json.load(open(path))
```

Comment:
You're loading as json here and then dumping as a string in `gcp`, you could instead pass the raw data and upload it directly.

Reply:
Done.